tests/adapters/test_record_adapter.py

```diff
@@ -155,3 +155,42 @@ def test_filters_or_records_validation(mock_record):
     with pytest.raises(ValueError):
         validate_filters_or_records([mock_record, ('Number', 'equals', 1)])
 
+
+def test_bulk_modify(mock_swimlane, mock_app, mock_record):
+    mock_swimlane._Swimlane__settings = {
+                'apiVersion': '2.17.0'
+            }
+    with mock.patch.object(mock_swimlane, 'request', return_value=None) as mock_func:
+        mock_app.records.bulk_modify(('Numeric', 'equals', 1), values={'Numeric': 2})
+    mock_func.assert_called_once_with('put', "app/{0}/record/batch".format(mock_app.id),json=
+    {'filters': [{
+            'fieldId': 'aqkg3', 'filterType': 'equals', 'value': 1
+        }],
+      'modifications': [{
+                'fieldId': {
+                    'type': 'Id',
+                    'value': 'aqkg3'
+                },
+                'type': 'Create', 'value': 2
+            }]
+    })
+    with mock.patch.object(mock_swimlane, 'request', return_value=None) as mock_func:
```

Comment:
Comments for each of these `with` block groups for what individual piece they're testing. Can split them into smaller individual tests with more descriptive names to help organize and report on any actual failures.